### PR TITLE
Add new mypy config `warn_unreachable`.

### PR DESCRIPTION
**Pull request by hukkinj1**
1. Run `pre-commit autoupdate`.
2. Add new mypy config `warn_unreachable`.
3. Fix the warnings for unreachable code

### CODE REVIEW

File: tests/test_basics.py [hunk 1]
```diff
@@ -164,19 +164,6 @@ def test_func_with_recursive_data_within(data):
 
 
 class TestDecoration(unittest.TestCase):
-    def test_decorate_module(self):
```
Comment (hukkinj1) [inline tests/test_basics.py]: These two tests are testing an error on the user's side. They are trying to decorate  a non-class and non-callable object. Our typehint for `__call__` says we don't support it, so also IMO shouldn't build handling for it or test it.

File: loggo/loggo.py [hunk 2]
```diff
@@ -252,9 +248,11 @@ def full_decoration(*args: Any, **kwargs: Any) -> Any:
             Args and kwargs are for/from the decorated function
             """
             bound = self._params_to_dict(function, *args, **kwargs)
-            # bound will be none if inspect signature binding failed. in this
```
Comment (hukkinj1) [inline loggo/loggo.py]: This comment was wrong. `bound` could never be `None`.

File: loggo/loggo.py [hunk 3]
```diff
@@ -252,9 +248,11 @@ def full_decoration(*args: Any, **kwargs: Any) -> Any:
...
-            # case, error log was created, raised if self.raise_logging_errors
```
Comment (hukkinj1) [inline loggo/loggo.py]: Also this was wrong. No error log was created.

File: loggo/loggo.py [hunk 4]
```diff
@@ -252,9 +248,11 @@ def full_decoration(*args: Any, **kwargs: Any) -> Any:
...
             if bound is None:
```
Comment (ilkericyuz) [inline loggo/loggo.py]: if `bound` could never be `None`, why this check?

Comment (hukkinj1) [reply loggo/loggo.py]: Probably a bug. But now it can be `None` with the changes made to `_params_to_dict`.

**Pull request merged**